src/lib.rs

```diff
@@ -278,6 +280,64 @@ impl From<Ulid> for u128 {
     }
 }
 
+impl From<[u8; 16]> for Ulid {
+    fn from(bytes: [u8; 16]) -> Self {
+        let value = u128::from(bytes[0]) << 120
+            | u128::from(bytes[1]) << 112
+            | u128::from(bytes[2]) << 104
+            | u128::from(bytes[3]) << 96
+            | u128::from(bytes[4]) << 88
+            | u128::from(bytes[5]) << 80
+            | u128::from(bytes[6]) << 72
+            | u128::from(bytes[7]) << 64
+            | u128::from(bytes[8]) << 56
+            | u128::from(bytes[9]) << 48
+            | u128::from(bytes[10]) << 40
+            | u128::from(bytes[11]) << 32
+            | u128::from(bytes[12]) << 24
+            | u128::from(bytes[13]) << 16
+            | u128::from(bytes[14]) << 8
+            | u128::from(bytes[15]);
+
+        Self(value)
+    }
+}
+
+impl From<Ulid> for [u8; 16] {
+    fn from(ulid: Ulid) -> Self {
+        let value = ulid.0;
+
+        [
+            ((value >> 120) & 0xFF) as u8,
+            ((value >> 112) & 0xFF) as u8,
+            ((value >> 104) & 0xFF) as u8,
+            ((value >> 96) & 0xFF) as u8,
+            ((value >> 88) & 0xFF) as u8,
+            ((value >> 80) & 0xFF) as u8,
+            ((value >> 72) & 0xFF) as u8,
+            ((value >> 64) & 0xFF) as u8,
+            ((value >> 56) & 0xFF) as u8,
+            ((value >> 48) & 0xFF) as u8,
+            ((value >> 40) & 0xFF) as u8,
+            ((value >> 32) & 0xFF) as u8,
+            ((value >> 24) & 0xFF) as u8,
+            ((value >> 16) & 0xFF) as u8,
+            ((value >> 8) & 0xFF) as u8,
+            (value & 0xFF) as u8,
+        ]
+    }
+}
+
+impl TryFrom<&[u8]> for Ulid {
```

Comment:
Remove in favor of letting the caller assert the length. We can add docs similar to <https://doc.rust-lang.org/nightly/std/primitive.u128.html#method.from_be_bytes>.

Reply:
Removed, where do we want the docs for this one,  `from_bytes` ?

Reply:
That would be a good spot, going to merge as-is though.